src/test/clojure/clara/test_rules.clj

```diff
@@ -5187,3 +5260,30 @@
                         (fire-rules)
                         (query temperature-query)))))
 
+(deftest test-stored-insertion-retraction-ordering
```

Comment:
can we add a more robust somewhere for these?  testing multiple insertions/retractions prior to firing rules and then repeating multiple insertions and retractions is what I'd be looking for.